gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/usecase/ApiGetDocumentationPagesUsecase.java

```diff
@@ -15,23 +15,55 @@
  */
 package io.gravitee.apim.core.documentation.usecase;
 
+import io.gravitee.apim.core.documentation.crud_service.PageCrudService;
 import io.gravitee.apim.core.documentation.domain_service.ApiDocumentationDomainService;
+import io.gravitee.apim.core.documentation.model.Breadcrumb;
 import io.gravitee.apim.core.documentation.model.Page;
+import io.gravitee.rest.api.service.exceptions.PageNotFoundException;
+import java.util.ArrayList;
+import java.util.LinkedList;
 import java.util.List;
+import org.apache.commons.lang3.StringUtils;
 
 public class ApiGetDocumentationPagesUsecase {
 
     private final ApiDocumentationDomainService apiDocumentationDomainService;
+    private final PageCrudService pageCrudService;
 
-    public ApiGetDocumentationPagesUsecase(ApiDocumentationDomainService apiDocumentationDomainService) {
+    public ApiGetDocumentationPagesUsecase(ApiDocumentationDomainService apiDocumentationDomainService, PageCrudService pageCrudService) {
         this.apiDocumentationDomainService = apiDocumentationDomainService;
+        this.pageCrudService = pageCrudService;
     }
 
     public Output execute(Input input) {
-        return new Output(apiDocumentationDomainService.getApiPages(input.apiId));
+        var pages = apiDocumentationDomainService.getApiPages(input.apiId, input.parentId);
+
+        List<Breadcrumb> breadcrumbs = new ArrayList<>();
+
+        if (!"ROOT".equals(input.parentId)) {
+            var pagesBreadcrumbs = this.constructBreadcrumbs(input.parentId, new LinkedList<>());
+
+            for (int i = 0; i < pagesBreadcrumbs.size(); i++) {
+                var page = pagesBreadcrumbs.get(i);
+                breadcrumbs.add(Breadcrumb.builder().id(page.getId()).name(page.getName()).position(i + 1).build());
+            }
+        }
+
+        return new Output(pages, breadcrumbs);
     }
 
-    public record Input(String apiId) {}
+    public record Input(String apiId, String parentId) {}
+
+    public record Output(List<Page> pages, List<Breadcrumb> breadcrumbs) {}
 
-    public record Output(List<Page> pages) {}
+    private LinkedList<Page> constructBreadcrumbs(String parentId, LinkedList<Page> breadcrumbs) {
+        if (!StringUtils.isEmpty(parentId)) {
+            try {
+                final Page optParent = pageCrudService.get(parentId);
```

Comment:
Suggestion: add a method `find` in pageCrudService that doesn't throw exception but instead returns an Optional.